### PR TITLE
[merged] push: use credentials in skopeo copy

### PR DESCRIPTION
**Pull request by runcom**
If you don't have any credentials stored within `~/.docker/config.json` for a given docker registry, `atomic` prompts the user for username and password.
It does a `docker login` after that, but `docker-py` isn't updating the `~/.docker/config.json` config file - this will result in `skopeo copy` failing to upload to the given docker registry (since it doesn't have any auth token at this point!).
```sh
$ sudo atomic push docker.io/runcom/busybox
Registry Username: runcom
Registry Password:
Uploading blob sha256:e02e811dd08fd49e7f6032625495118e63f597eb150403d02e3238af1df240ba
 0 B / 1.43 KB [---------------------------------------------------------------]
Uploading blob sha256:e88b3f82283bc59d5e0df427c824e9f95557e661fcb0ea15fb0fb6f97760f9d9
 8 B / 1.23 MB [>--------------------------------------------------------------]
FATA[0007] Error writing blob: Error initiating layer upload to runcom/busybox/blobs/uploads/, status 401
```
I'm not sure why `atomic` does a docker login at this point since it's not storing any auth token to be used by `skopeo copy`.
This patch fix this whole story by passing down to `skopeo` the username and password and using a new `skopeo` flag (`--dest-creds`) in order to fix this behavior.
```
$ sudo atomic push docker.io/runcom/busybox:latest
Registry Username: runcom
Registry Password:
Copying blob sha256:e88b3f82283bc59d5e0df427c824e9f95557e661fcb0ea15fb0fb6f97760f9d9
 1.23 MB / 1.23 MB [===========================================================]
Copying config sha256:e02e811dd08fd49e7f6032625495118e63f597eb150403d02e3238af1df240ba
 0 B / 1.43 KB [---------------------------------------------------------------]
Writing manifest to image destination
Storing signatures 
```

@baude @rhatdan PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
This will of course required a skopeo bump in (at least) Fedora - /cc @lsm5

File: Atomic/util.py [hunk 1]
```diff
@@ -347,6 +347,9 @@ def skopeo_copy(source, destination, debug=False, sign_by=None, insecure=False,
     if insecure:
         cmd = cmd + ['--tls-verify=false']
     cmd = cmd + ['copy']
+    if username:
+        # it's ok to send an empty password (think of krb for instance)
+        cmd = cmd + [ "--dest-creds=%s:%s" % (username, password) ]
```
Comment (rhatdan) [inline Atomic/util.py]: This will blow up if password=None

Comment (runcom) [inline Atomic/util.py]: I think I figured

File: Atomic/util.py [hunk 2]
```diff
@@ -347,6 +347,9 @@ def skopeo_copy(source, destination, debug=False, sign_by=None, insecure=False,
     if insecure:
         cmd = cmd + ['--tls-verify=false']
     cmd = cmd + ['copy']
+    if username:
+        # it's ok to send an empty password (think of krb for instance)
+        cmd = cmd + [ "--dest-creds=%s:%s" % (username, password if password else "") ]
```
Comment (rhatdan) [inline Atomic/util.py]: Does this work better

```
  cmd = cmd + [ "--dest-creds=%s%s" % (username, ":%s" % password if password else "") ]
```

Comment (rhatdan) [inline Atomic/util.py]: Don't want ":" if no password

**Comment by runcom** [comment]
Rebased

**Comment by runcom** [comment]
@rhatdan this is now rebased

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**